web/test/spec/views/record/list-controller.spec.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+describe('driver.views.record: ListController', function () {
+
+    beforeEach(module('ase.mock.resources'));
+    beforeEach(module('driver.mock.resources'));
+    beforeEach(module('driver.views.record'));
+
+    var $controller;
+    var $httpBackend;
+    var $rootScope;
+    var $scope;
+    var Controller;
+    var DriverResourcesMock;
+    var ResourcesMock;
+
+    // Initialize the controller and a mock scope
+    beforeEach(inject(function (_$controller_, _$httpBackend_, _$rootScope_,
+                                _DriverResourcesMock_, _ResourcesMock_) {
+        $controller = _$controller_;
+        $httpBackend = _$httpBackend_;
+        $rootScope = _$rootScope_;
+        $scope = $rootScope.$new();
+        DriverResourcesMock = _DriverResourcesMock_;
+        ResourcesMock = _ResourcesMock_;
+    }));
+
+    it('should have header keys', function () {
+        var recordType = ResourcesMock.RecordType;
+        var recordTypeId = recordType.uuid;
+        var recordTypeIdUrl = new RegExp('api/recordtypes/' + recordTypeId);
+        $httpBackend.expectGET(recordTypeIdUrl).respond(200, recordType);
+
+        var recordSchema = ResourcesMock.RecordSchema;
+        var recordSchemaId = recordSchema.uuid;
+        var recordSchemaIdUrl = new RegExp('api/recordschemas/' + recordSchemaId);
+        $httpBackend.expectGET(recordSchemaIdUrl).respond(200, recordSchema);
+
+        var recordResponse = DriverResourcesMock.RecordResponse;
+        var recordsByTypeUrl = new RegExp(recordTypeId);
+        $httpBackend.expectGET(recordsByTypeUrl).respond(200, recordResponse);
+        /*
+         TODO: can't figure out why this isn't working:
+         var recordsByTypeUrl = new RegExp('api/records/?record_type=' + recordTypeId);
```

Comment:
Huh, it looks ok and first glance, and I would need to play with it myself to fix it, but two things about that error message stand out.
1. The hostname is not on the 'expected' response
2. There is a trailing slash on the expected response
Based on the regex you're providing though, neither of those things should matter.

The only special characters in that regex are the slash and the question mark.